text/0000-iri-v181-transaction-and-bundle/0000-iri-v181-transaction-and-bundle.md

```diff
@@ -0,0 +1,980 @@
++ Feature name: `iri_v181_transaction_and_bundle`
++ Start date: 2019-09-06
++ RFC PR: [iotaledger/bee-rfcs#3](https://github.com/iotaledger/bee-rfcs/pull/3),
+  [iotaledger/bee-rfcs#18](https://github.com/iotaledger/bee-rfcs/pull/18),
+  [iotaledger/bee-rfcs#20](https://github.com/iotaledger/bee-rfcs/pull/20)
++ Bee issue: [iotaledger/bee#0000](https://github.com/iotaledger/bee/issues/0000)
+
+# Summary
+
+The fundamental communication unit in the IOTA protocol is the transaction. Messages, including payment settlements and
+plain data, are propagated through the IOTA network in transactions. Message that are too large are split up into bundles
+of several transactions and sent one by one.
+
+This RFC proposes a `Transaction` type and a `Bundle` type to represent the transaction and bundle formats used by the
+IOTA Reference Implementation as of release [`iri v1.8.1`]. We also propose the `TransactionBuilder`,
+`IncomingBundleBuilder`, and `OutgoingBundleBuilder` types following the builder pattern to construct both message
+types, respectively. We distinguish between incoming and outgoing bundles because one is concerned with verifying an
+existing message being received, while the other one is concerned with constructing a new message intended to be sent,
+which includes setting a number of fields like the bundle hash.
+
+Useful links:
+
++ [Trinary](https://docs.iota.org/docs/dev-essentials/0.1/concepts/trinary)
++ [What is a transaction?](https://docs.iota.org/docs/getting-started/0.1/introduction/what-is-a-transaction)
++ [What is a bundle?](https://docs.iota.org/docs/getting-started/0.1/introduction/what-is-a-bundle)
++ [Bundles and transactions](https://docs.iota.org/docs/dev-essentials/0.1/concepts/bundles-and-transactions)
++ [Structure of a transaction](https://docs.iota.org/docs/dev-essentials/0.1/references/structure-of-a-transaction)
++ [Structure of a bundle](https://docs.iota.org/docs/dev-essentials/0.1/references/structure-of-a-bundle)
++ [Security levels](https://docs.iota.org/docs/dev-essentials/0.1/references/security-levels)
+
+# Motivation
+
+IOTA is a transaction settlement and data transfer layer for the Internet of Things. Messages on the network are
```

Comment:
we should not restrict ourselves to IoT. IOTA should also be for the classic Internet.

Reply:
I think the term IOT already includes classic internet and much boarder meaning.

Reply:
We usually differentiate IoT and Internet though, I agree with @Alex6323, I'm not sure we should emphasise on IoT here.

Reply:
In general, IoT consists of end devices (sensors and actuators), gateways, and the clouds (including internet). The applications of an IoT system usually include sensing, data analytics, decision making, and actions. So regarding IOTA as a data transfer layer for IoT makes sense.

Reply:
To me, the term IoT also includes the general internet. At the end of the day, in IoT you are also sending information from full sized computers to embedded devices and the other way round.